Snippets of Pytest output before and after this change (`python -m pytest -vvk longest` with a failing unit test):

# Before

```
FAILED tests/pathfinding/test_single_pair_longest_path_core.py::test_splp_smoketest[cache-dijkstra] - assert [<edgegraph.structure.vertex.Vertex object at 0x7fcb389694f0>, <edgegraph.structure.vertex.Vertex object at 0x7fcb387d8600>, <edgegraph.structure.vertex.Vertex object at 0x7fcb387e2e60>, <edgegraph.structure.vertex.Vertex object at 0x7fcb389695e0>, <edgegraph.structure.vertex.Vertex object at 0x7fcb388492b0>, <edgegraph.structure.vertex.Vertex object at 0x7fcb3884c590>, <edgegraph.structure.vertex.Vertex object at 0x7fcb3884c4d0>] == [<edgegraph.structure.vertex.Vertex object at 0x7fcb389694f0>, <edgegraph.structure.vertex.Vertex object at 0x7fcb387d8600>, <edgegraph.structure.vertex.Vertex object at 0x7fcb387e2e60>, <edgegraph.structure.vertex.Vertex object at 0x7fcb389695e0>, <edgegraph.structure.vertex.Vertex object at 0x7fcb388492b0>, <edgegraph.structure.vertex.Vertex object at 0x7fcb3884c590>, <edgegraph.structure.vertex.Vertex object at 0x7fcb3884c590>]
  
  At index 6 diff: <edgegraph.structure.vertex.Vertex object at 0x7fcb3884c4d0> != <edgegraph.structure.vertex.Vertex object at 0x7fcb3884c590>
  
  Full diff:
    [
        <edgegraph.structure.vertex.Vertex object at 0x7fcb389694f0>,
        <edgegraph.structure.vertex.Vertex object at 0x7fcb387d8600>,
        <edgegraph.structure.vertex.Vertex object at 0x7fcb387e2e60>,
        <edgegraph.structure.vertex.Vertex object at 0x7fcb389695e0>,
        <edgegraph.structure.vertex.Vertex object at 0x7fcb388492b0>,
        <edgegraph.structure.vertex.Vertex object at 0x7fcb3884c590>,
  -     <edgegraph.structure.vertex.Vertex object at 0x7fcb3884c590>,
  ?                                                             ^^
  +     <edgegraph.structure.vertex.Vertex object at 0x7fcb3884c4d0>,
  ?                                                             ^^
    ]
============================= 2 failed, 828 deselected in 0.62s =============================
```

# After

```
FAILED tests/pathfinding/test_single_pair_longest_path_core.py::test_splp_smoketest[cache-dijkstra] - assert [<Vertex i=1 @ 0x7fd6526e94f0>, <Vertex i=4 @ 0x7fd652758600>, <Vertex i=8 @ 0x7fd652762e60>, <Vertex i=0 @ 0x7fd6526e95e0>, <Vertex i=2 @ 0x7fd6525cd2b0>, <Vertex i=5 @ 0x7fd656fa84d0>, <Vertex i=6 @ 0x7fd6525d0590>] == [<Vertex i=1 @ 0x7fd6526e94f0>, <Vertex i=4 @ 0x7fd652758600>, <Vertex i=8 @ 0x7fd652762e60>, <Vertex i=0 @ 0x7fd6526e95e0>, <Vertex i=2 @ 0x7fd6525cd2b0>, <Vertex i=5 @ 0x7fd656fa84d0>, <Vertex i=5 @ 0x7fd656fa84d0>]
  
  At index 6 diff: <Vertex i=6 @ 0x7fd6525d0590> != <Vertex i=5 @ 0x7fd656fa84d0>
  
  Full diff:
    [
        <Vertex i=1 @ 0x7fd6526e94f0>,
        <Vertex i=4 @ 0x7fd652758600>,
        <Vertex i=8 @ 0x7fd652762e60>,
        <Vertex i=0 @ 0x7fd6526e95e0>,
        <Vertex i=2 @ 0x7fd6525cd2b0>,
        <Vertex i=5 @ 0x7fd656fa84d0>,
  -     <Vertex i=5 @ 0x7fd656fa84d0>,
  ?               ^          ^^^^^
  +     <Vertex i=6 @ 0x7fd6525d0590>,
  ?               ^          ^^ +++
    ]
============================= 2 failed, 828 deselected in 0.62s =============================
```

--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -20,14 +20,20 @@ from .fixtures import *  # noqa: F403
 
 LOG = logging.getLogger(__name__)
 
+
 # monkey-patch Vertex to pretty-repr itself with the `i` attribute if
 # available, since so many unit tests use this pattern.  Makes for MUUUUCH
 # easier debugging; but I don't want to make this a global API feature of
 # Vertex -- so for the unittests only!
 def new_repr(inst):
-    if hasattr(inst, 'i') and isinstance(inst.i, int):
+    """
+    Monkey-patched __repr__ for Vertex to print `i` attribute.
+    """
+    if hasattr(inst, "i") and isinstance(inst.i, int):
         return f"<Vertex i={inst.i} @ {hex(id(inst))}>"
     return super(Vertex, inst).__repr__()
+
+
 Vertex.__repr__ = new_repr
 
 

